lib/mpool/src/mpool.c

```diff
@@ -357,41 +357,43 @@ mpool_mclass_props_get(struct mpool *mp, enum mpool_mclass mclass, struct mpool_
 {
     struct media_class *mc;
 
-    if (!mp || mclass >= MP_MED_COUNT)
+    if (!mp || mclass >= MP_MED_COUNT || !props)
         return merr(EINVAL);
 
+    memset(props, 0, sizeof(*props));
+
     mc = mp->mc[mclass];
     if (!mc)
         return merr(ENOENT);
 
-    if (props)
-        props->mc_mblocksz = mclass_mblocksz_get(mc) >> MB_SHIFT;
+    const struct mblock_fset *mbfsp = mclass_fset(mc);
```

Comment:
mpool.c is totally unaware of what a fileset is. It's managed by the mclass layer. So, it is cleaner for mpool.c to go via the mclass interfaces to access fmaxsz and filecnt.